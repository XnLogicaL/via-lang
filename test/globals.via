## This file is a part of the via Programming Language project
## Copyright (C) 2024-2025 XnLogical - Licensed under GNU GPL v3.0

## Global declaration
global x = 10
print(x)

## Global arithmetics
global y = 5
global result = x + y
print(result)

## Scoped variables
do {
  global n = 15
  print(n)
}

## Global types
global intVar: int = 10
print(type intVar)
print(intVar)

global floatVar: float = 1.0
print(type floatVar)
print(floatVar)

global boolVar: bool = true
print(type boolVar)
print(boolVar)

global stringVar: string = "Hello"
print(type stringVar)
print(stringVar)

## nameof
global thisIsACoolName = 10
print(nameof thisIsACoolName)