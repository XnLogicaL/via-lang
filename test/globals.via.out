10
15
15
int
10
float
1.000000
bool
true
string
Hello
thisIsACoolName